vulkan_generator/vulkan_parser/parser.py

```diff
@@ -16,11 +16,61 @@
 
 
 from pathlib import Path
+import sys
 
 from vulkan_generator.vulkan_parser.api import types
-from vulkan_generator.vulkan_parser.internal import parser as vulkan_parser
+from vulkan_generator.vulkan_parser.internal import parser as internal_parser
 from vulkan_generator.vulkan_parser.postprocess import postprocess
 
 
-def parse(filename: Path) -> types.VulkanInfo:
-    return postprocess.process(vulkan_parser.parse(filename))
+def print_vulkan_metadata(vulkan_metadata: types.VulkanInfo) -> None:
+    """Prints all the vulkan information that is extracted"""
+
+    print(vulkan_metadata.platforms)
+    print(vulkan_metadata.includes)
+    print(vulkan_metadata.defines)
+
+    vulkan_types = vulkan_metadata.types
+    print(vulkan_types.handles)
+    print(vulkan_types.handle_aliases)
+    print(vulkan_types.bitmasks)
+    print(vulkan_types.bitmask_aliases)
+    print(vulkan_types.enums)
+    print(vulkan_types.enum_aliases)
+    print(vulkan_types.structs)
+    print(vulkan_types.struct_aliases)
+    print(vulkan_types.funcpointers)
+
+    vulkan_commands = vulkan_metadata.commands
+    print(vulkan_commands.commands)
+    print(vulkan_commands.command_aliases)
+
+    print(vulkan_metadata.core_versions)
+    print(vulkan_metadata.extensions)
+    print(vulkan_metadata.image_formats)
+
+    spirv_metadata = vulkan_metadata.spirv_metadata
+    print(spirv_metadata.extensions)
+    print(spirv_metadata.capabilities)
+
+
+def parse(filename: Path, dump: bool = False) -> types.VulkanInfo:
+    metadata = postprocess.process(internal_parser.parse(filename))
+
+    if dump:
+        print_vulkan_metadata(metadata)
+
+    return metadata
+
+
+if __name__ == "__main__":
+    if len(sys.argv) == 2:
+        parse(Path(sys.argv[1]))
+        sys.exit(0)
+
+    if len(sys.argv) == 3:
+        if sys.argv[2].strip().lower() == "dump":
+            parse(Path(sys.argv[1]), True)
+            sys.exit(0)
+
+    print("Please use as <xml location> Optional[<dump>]")
```

Comment:
<xml location> implies "substitute location here". <dump> means "write 'dump' or do not write 'dump'". Maybe you should choose some way of communicating this that shows the difference. I'd suggest Optional['dump']

Reply:
Sure.